test/grabURITest.coffee

```diff
@@ -0,0 +1,205 @@
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+    it 'should match when using domain name without ftp', ->
+      urlRepresentation = 'ftp://example.com'
+
+      room.user.say('user1', "#{urlRepresentation}").then =>
+        expect(room.messages).to.eql [
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+    it 'should match FTPs containing individual pages', ->
+      urlRepresentation = 'ftp://ftp.example.com/index.zip'
+
+      room.user.say('user1', "#{urlRepresentation}").then =>
+        expect(room.messages).to.eql [
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+    it 'should match when using domain name without ftp', ->
+      urlRepresentation = 'ftp://example.com/index.html?file=requestedFileName.zip'
+
+      room.user.say('user1', "#{urlRepresentation}").then =>
+        expect(room.messages).to.eql [
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+    it 'should match when using domain name ftp with user password and decorator symbol', ->
+      urlRepresentation = 'ftp://user:password@host.private'
+
+      room.user.say('user1', "#{urlRepresentation}").then =>
+        expect(room.messages).to.eql [
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+    it 'should match when using domain name with tilda', ->
+      urlRepresentation = 'https://www.cs.tut.fi/~jkorpela/ftpurl.html'
+
+      room.user.say('user1', "#{urlRepresentation}").then =>
+        expect(room.messages).to.eql [
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+    it 'should match when using domain name with anchor', ->
+      urlRepresentation = 'http://fullurlhere.com/nbs-test-panel-of-diseases-2#anchorlink1'
+
+      room.user.say('user1', "#{urlRepresentation}").then =>
+        expect(room.messages).to.eql [
+          [ 'user1', "#{urlRepresentation}" ]
+          [ 'hubot', "Meow, Uploading: #{urlRepresentation}" ]
+        ]
+
+  context 'Server Status Not 200', ->
+    room = null
+
+    beforeEach ->
+      room = helper.createRoom()
+      do nock.disableNetConnect
+      nock('http://fakeAPIwebsite.com')
+        .post('/url')
+        .replyWithError('something awful happened');
+
+    afterEach ->
+      room.destroy()
+      nock.cleanAll()
+
+    it 'should reply to User when handed proper URL', ->
+      room.user.say('user1', 'http://fakewebsite.com').then =>
+        console.log room.messages
+        expect(room.messages).to.eql [
+          [ 'user1', 'http://fakewebsite.com' ]
+          [ 'hubot', "Meow, Uploading: http://fakewebsite.com" ]
+        ]
```

Comment:
I don't think I can use Nock to test this. The HTTP server created by the Hubot Helper is self contained.